### ISSUE TITLE
Developer data without developer_data_id message

### ISSUE THREAD

**Issue by AartGoossens** [dtcooper/python-fitparse#124]
Hi,

I have some FIT files that raise a `FitParseError` because they contains developer data (or more specifically: "field description" messages) without the "developer_data_id" message.

I cannot share the files because the data is not mine but I attached a dump of the "field_description" messages in a FIT file coming from a Garmin 235: [field_descriptions.txt](https://github.com/dtcooper/python-fitparse/files/6334858/field_descriptions.txt)
There are no "developer_data_id" messages in this FIT file.

The full traceback of the exception is:
```
Traceback (most recent call last):
  File "script.py", line 10, in <module>
    for record in fitfile.get_messages():
  File "/home/aart/projects/python-fitparse/fitparse/base.py", line 497, in get_messages
    for message in super(CacheMixin, self).get_messages(names, with_definitions, as_dict):
  File "/home/aart/projects/python-fitparse/fitparse/base.py", line 467, in get_messages
    message = self._parse_message()
  File "/home/aart/projects/python-fitparse/fitparse/base.py", line 483, in _parse_message
    self._messages.append(super(CacheMixin, self)._parse_message())
  File "/home/aart/projects/python-fitparse/fitparse/base.py", line 168, in _parse_message
    add_dev_field_description(message)
  File "/home/aart/projects/python-fitparse/fitparse/records.py", line 461, in add_dev_field_description
    raise FitParseError("No such dev_data_index=%s found" % (dev_data_index))
fitparse.utils.FitParseError: No such dev_data_index=0 found
```

It is my understanding that this FIT file is not according to spec (i.e. there should always be a "developer_data_id" message before the "field_description" field) but I want inform whether you are interested in handling these errors more gracefully.
As a side note: The Java implementation that is part of the Garmin FIT SDK download seems to handle these files without problem.

I think there are 2 levels at which this can be handled:
1. Allow these errors to be ignored by setting e.g. a `check_developer_data_id=False` kwarg similar to `check_crc=`. This will get rid of the "field_description" messages and developer data. This would happen in `base.py#FitFileDecoder._parse_message()`.
2. Allow "field_description" messages for which there has not been a "developer_data_id" message to be stored anyway by creating the entry in `DEV_DATA` on the fly, setting the "application id" to `None` as it is not provided. This would happen in `records.py#add_dev_field_description()`.
3. For developer data for which there is no "developer_data_id" and no "field_description" a similar approach as (2) could be taken, creating the relevant entry in `DEV_DATA` on the fly. This would happen in `records.py#get_dev_type()`.

(2) and (3) can be implemented without (1) (maybe raising a warning) but as this is not according to FIT spec I assume that is a no-go.

If there is support for this feature I am willing to work on a PR.

Edit: I attached the wrong traceback.

**Comment by AartGoossens** [comment]
I have been working on an implementation to solve this issue in my fork. Relevant commit here: https://github.com/AartGoossens/python-fitparse/commit/ce09b5c09824f54fd04826c60d53f6c78b47212b
It basically enables you to pass a `check_developer_data` kwarg to the `FitFile()` init:
```python
fitfile = FitFile("fit/file/with/erroneous/developer/data.fit", check_developer_data=False)
```
It will print warnings when the developer data is not initialized correctly.

This code change solves the issue that I had.

Let me know if you are interested in merging in a code change like this. I am willing to work on alternative solutions as well.

**Comment by dtcooper** [comment]
PR welcome!

**Comment by AartGoossens** [comment]
Thanks.
As my fix is built on top of the changes in #126 I will create a PR once we finished that PR, to prevent duplicate work.
